Sources/Sentry/SentryClient.m

```diff
@@ -422,6 +433,35 @@ - (void)setUserIdIfNoUserSet:(SentryEvent *)event
     }
 }
 
+/**
+ * Calls onCrashedLastRun if the execution terminated with a crash.
+ */
+- (void)callOnCrashedLastRun:(SentryEvent *)event
+{
+    if (nil != self.options.onCrashedLastRun && self.crashAdapter.crashedLastLaunch
+        && !SentrySDK.crashedLastRunCalled && nil != event.exceptions) {
+
+        NSPredicate *unhandledExpeptions = [NSPredicate predicateWithBlock:^BOOL(
+            id _Nullable object, NSDictionary<NSString *, id> *_Nullable bindings) {
+            SentryException *exception = object;
+            return nil != exception.mechanism && nil != exception.mechanism.handled
+                && ![exception.mechanism.handled boolValue];
+        }];
```

Comment:
H: I think this would work for most of the cases, but it's flakey IMO.
how do we know that the `event` calling `onCrashedLastRun` is actually the one that caused the crash?

I'd suggest another approach:

SentryCrash sets `fixture.crashAdapter.internalCrashedLastLaunch` so it knows the event that caused this flag to be toggled.
what's about caching the `eventId` as well that caused the App. to be crashed and later, we filter by eventId? so we'll be sure that the `onCrashedLastRun` is actually caled with the `event` that actually crashed the App.

My concern is, if the App. is offline and there are multiple and distinct hard crashes, the event called `onCrashedLastRun` might not be the event that actually caused the last crash and the feedback won't be accurate.

Reply:
It's very good that you think about such edge cases, but I think that your concern might not be worth the effort. Let me explain why.

When a crash happens, SentryCrash writes a crash-report.json to disk. The next time the app is launched, the SDK installs the `SentryCrashIntegration`, which calls the [`SentryCrashReportConverter`](https://github.com/getsentry/sentry-cocoa/blob/8a9faea4e8863fecc1dfe2e8094c8d6ee16ab524/Sources/Sentry/SentryCrashReportConverter.m#L80-L124), who converts the crash-report.json to a SentryEvent. The [`SentryCrashReportSink`](https://github.com/getsentry/sentry-cocoa/blob/feat/on-crashed-last-run/Sources/Sentry/SentryCrashReportSink.m#L21) then calls `[SentrySDK captureCrashEvent:event];`. At that point, SentrySDK knows that it is an event with a crash. The [`SentryHub`](https://github.com/getsentry/sentry-cocoa/blob/8a9faea4e8863fecc1dfe2e8094c8d6ee16ab524/Sources/Sentry/SentryHub.m#L191-L210) also has a `captureCrashEvent`, but the `SentryClient` doesn't. Therefore I suggest that the `SentryHub` just passes this information to the `SentryClient` and in `callOnCrashedLastRun` we can simplify the logic a lot. 

Good point I'm going to change this.